### PR TITLE
Implement `LocationAware` for `FileLocationPatternConverter` (#2781)

### PR DESCRIPTION
**Pull request by dedeibel**
Fix for https://github.com/apache/logging-log4j2/issues/2781

_See also https://github.com/apache/logging-log4j2/pull/2782 (main branch)_

### CODE REVIEW

File: src/changelog/.2.x.x/2781_fix_file_location_pattern_converter_location_requirement.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<entry xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns="https://logging.apache.org/xml/ns"
+       xsi:schemaLocation="https://logging.apache.org/xml/ns https://logging.apache.org/xml/ns/log4j-changelog-0.xsd"
+       type="fixed">
+  <issue id="2781" link="https://github.com/apache/logging-log4j2/issues/2781"/>
+  <description format="asciidoc">Fix handling of location requirement for pattern key `%F` / `%file`.</description>
```
Comment (vy) [inline src/changelog/.2.x.x/2781_fix_file_location_pattern_converter_location_requirement.xml:7]:
```suggestion
  <description format="asciidoc">Fix location requirement for the `%F` and `%file` keys in Pattern Layout</description>
```

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/pattern/LocationPatternConvertersRequireLocationTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+import org.apache.logging.log4j.core.impl.LocationAware;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class LocationPatternConvertersRequireLocationTest {
```
Comment (vy) [inline log4j-core-test/src/test/java/org/apache/logging/log4j/core/pattern/LocationPatternConvertersRequireLocationTest.java:29]:
```suggestion
class LocationPatternConvertersRequireLocationTest {
```

Comment (vy) [reply log4j-core-test/src/test/java/org/apache/logging/log4j/core/pattern/LocationPatternConvertersRequireLocationTest.java:29]: Make tests package private

**Review by vy — APPROVED** [review]
@dedeibel, good job! :100:

@ppkarwasz, I think this can be merged. Since you're the one who first reacted to the issue, I'd appreciate your conclusive review. If you're okay with it, please merge the PR.

**Comment by vy** [comment]
@dedeibel, could you make sure `./mvnw verify -am -pl :log4j-core,:log4j-core-test` passes and do the same for #2782 too, please?

**Comment by dedeibel** [comment]
> @dedeibel, could you make sure `./mvnw verify -pl :log4j-core,:log4j-core-test` passes and do the same for #2782 too, please?

@vy I am sorry! It should be fixed now.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/package-info.java [hunk 3]
```diff
@@ -18,7 +18,7 @@
  * Provides classes implementing format specifiers in conversion patterns.
  */
 @Export
-@Version("2.21.1")
+@Version("2.22.0")
```
Comment (ppkarwasz) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/package-info.java:21]: _Nit_: When a minor version bump is required in the package, we usually bump it to version number of the next minor Log4j version (`2.24.0`).

This convention helps us remembering, in which Log4j version the package was modified last.

**Review by ppkarwasz — APPROVED** [review]
Looks good to me,

Thank you for you contribution.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/package-info.java (hunk 3 above)
Comment (dedeibel) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/package-info.java:21]: I see.

@vy should I bump it now again or better leave it at this point?

Comment (vy) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/package-info.java:21]: @ppkarwasz, you're right. I had blindly applied the `bnd-baseline:check` suggetion. Replacing it with `2.24.0`:

```suggestion
@Version("2.24.0")
```

**Comment by ppkarwasz** [comment]
Closes #2781

**Pull request merged**